src/Workspaces/CSharp/Portable/LanguageServices/CSharpTypeInferenceService.TypeInferrer.cs

```diff
@@ -263,8 +263,35 @@ private IEnumerable<ITypeSymbol> InferTypesWorker(int position)
                 var syntaxTree = (SyntaxTree)_semanticModel.SyntaxTree;
                 var token = syntaxTree.FindTokenOnLeftOfPosition(position, _cancellationToken);
                 token = token.GetPreviousTokenIfTouchingWord(position);
-
                 var parent = token.Parent;
+
+                // A strange broken code scenario:
+                // new Form
+                // {
+                //     Location = new $$
+                //     StartPosition = FormStartPosition.CenterParent
+                // };
+                // The 'new' token is part of an assignment of the assignment to StartPosition,
+                // but the user is really trying to assign to Location.
+                // In this case, we need to detect that the 'new' token is on the left side of 
+                // its parent assignment and on the right side of its grandparent assignment and
+                // return the inferred type of the grandparent.
+                if (token.Kind() == SyntaxKind.NewKeyword &&
+                    token.Parent.Kind() == SyntaxKind.ObjectCreationExpression &&
+                    token.Parent?.Parent.Kind() == SyntaxKind.SimpleAssignmentExpression)
```

Comment:
Though I suppose in theory there would be the CompilationUnitSyntax parent anyway, so maybe it's unncessary.